internal/config/config.go

```diff
@@ -0,0 +1,171 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+)
+
+//go:generate go run generate/main.go
+
+// Parsing functions.
+var (
+	toBool     = strconv.ParseBool
+	toInt      = strconv.Atoi
+	toInt64    = func(s string) (int64, error) { return strconv.ParseInt(s, 10, 64) }
+	toString   = func(s string) (string, error) { return s, nil }
+	toDuration = func(s string) (time.Duration, error) { return time.ParseDuration(s + "s") }
+	toLogLevel = func(s string) (LogLevel, error) {
+		var logLevel = map[string]LogLevel{
+			"debug":   LogLevelDebug,
+			"info":    LogLevelInfo,
+			"warning": LogLevelWarning,
+			"error":   LogLevelError,
+		}
+		if ll, ok := logLevel[s]; ok {
+			return ll, nil
+		} else {
+			return LogLevelDebug, fmt.Errorf(`invalid log level "%s"`, s)
+		}
+	}
+)
+
+func GetServerManagerSessionId() string {
+	return "default_session_id"
+}
+
+func GetInspectServerEndpoint() string {
+	port := getInspectServerPort()
+	if port == nil {
+		panic("")
+	}
+	return fmt.Sprintf("0.0.0.0:%d", *port)
+}
+
+func GetAuth() Auth {
+	// getting the (optional) account index
+	index := getTxSigningMnemonicAccountIndex()
+
+	// if the mnemonic is coming from an environment variable
+	if mnemonic := getTxSigningMnemonic(); mnemonic != nil {
+		return AuthMnemonic{*mnemonic, index}
+	}
+
+	// if the mnemonic is coming from a file
+	if file := getTxSigningMnemonicFile(); file != nil {
+		mnemonic, err := os.ReadFile(*file)
+		if err != nil {
+			panic(err)
+		}
+		return AuthMnemonic{string(mnemonic), index}
+	}
+
+	// if we are not using mnemonics, but AWS authentication
+	keyID := getTxSigningAwsKmsKeyId()
+	region := getTxSigningAwsKmsRegion()
+	if keyID == nil || region == nil {
+		panic("missing tx-signing environment variables")
+	}
+	return AuthAWS{*keyID, *region}
+}
+
+// ------------------------------------------------------------------------------------------------
+// Custom Types
+// ------------------------------------------------------------------------------------------------
+
+type LogLevel uint8
```

Comment:
Maybe create a `loglevel.go` file for the LogLevel type and the toLogLevel function.

Reply:
Created a `types` folder.